src/Framework/Telemetry/TaskExecutionStats.cs

```diff
@@ -5,11 +5,36 @@
 
 namespace Microsoft.Build.Framework;
 
-internal class TaskExecutionStats(TimeSpan cumulativeExecutionTime, short executionsCount, long totalMemoryConsumption)
+internal class TaskExecutionStats(TimeSpan cumulativeExecutionTime, int executionsCount, long totalMemoryConsumption)
 {
+    private TaskExecutionStats()
+        : this(TimeSpan.Zero, 0, 0)
+    { }
+
+    internal static TaskExecutionStats CreateEmpty()
```

Comment:
can it be 'Initialize'? 
The naming a bit unusual :)

Reply:
I think this is good naming. Initialize would imply: we first create and then initialize. This is a static factory method that returns an instance of stats that is empty. Adding a doc comment there